### PR TITLE
Begin implementing null backend

### LINKED ISSUE #28: To do: add a Null backend
It assumes the user already has an environment active with their required dependencies.

Discussed in #27.

### PR DESCRIPTION
**Pull request by kescobo**
If you'd rather deal with this yourself, fine with me if you want to just close this. Closes #28 

I'm not totally sure how to test this, since it would require getting the workflow runner to install the appropriate packages separately to make the other tests pass. But if you're ok with it, I think adding a disclaimer that this is untested would be ok :shrug:

### CODE REVIEW

File: src/backend.jl [hunk 1]
```diff
@@ -36,6 +38,9 @@ function conda_cmd(args=``; io::IO=stderr)
     b = backend()
     if b == :MicroMamba
         MicroMamba.cmd(args, io=io)
+    elseif b == :Null
+        @error "Backend is 'Null' - no action taken"
+        return nothing
```
Comment (cjdoris) [inline src/backend.jl:43]: I think just raise an error here. It's an internal function which should never be called with this backend anyway.

**Comment by kescobo** [comment]
- [x]  envdir - throw an error, and document it can throw?
- [x]  withenv - already covered by activate!, but could also return f()
  - The stuff before the `try` block isn't all that expensive - if it were me, I'd leave it to fall back to `activate!()`. But I put it in for now.
- [x]  which - just return Sys.which(progname)
- [ ]  resolve - immediately return true
  - Other parts of this function return `nothing`, I just added the check to the first conditional block.
- [x]  gc - do nothing
- [x]  status - needs some logic to avoid checking+printing the currently installed versions

**Comment by kescobo** [comment]
Oops - didn't mean to close that :laughing:

**Comment by cjdoris** [comment]
Looks good so far!
- Sure, I'm happy to put `withenv` back. The less special-casing the better.
- You're right about `resolve`, I was confusing it with something else.
- You added a message to `status` but there also needs some special casing to treat `Null` as always being resolved, but not actually displaying any installed versions. If you don't want to do that I'm happy to after the PR is merged.

It would be nice to test this, at least a bit. Maybe the tests could look at `JULIA_CONDAPKG_BACKEND` and modify the tests appropriately if it is Null?

File: src/resolve.jl [hunk 2]
```diff
@@ -197,8 +197,8 @@ function _run(io::IO, cmd::Cmd, args...; flags=String[])
 end
 
 function resolve(; force::Bool=false, io::IO=stderr, interactive::Bool=false, dry_run::Bool=false)
-    # if frozen, do nothing
-    if STATE.frozen
+    # if frozen or if backend is null, do nothing
+    if STATE.frozen || backend() == :Null
```
Comment (cjdoris) [inline src/resolve.jl:201]: When the backend is Null, we should also set `STATE.resolved = true`.

**Comment by cjdoris** [comment]
On `status`, specifically it needs to set `docurpkgs = resolved && backend() != :Null` then replace `resolved` with `docurpkgs` in most places.

**Comment by kescobo** [comment]
> On status, specifically it needs to set docurpkgs = resolved && backend() != :Null then replace resolved with docurpkgs in most places.

Here, I just have it return nothing up at the top after printing the message (that's what I originally intended, and just forgot :facepalm: ). Is that OK?

File: src/resolve.jl (hunk 2 above)
Comment (kescobo) [reply src/resolve.jl:201]: I put this into the `backend()` function, which seems like the best place, no?

**Comment by kescobo** [comment]
> It would be nice to test this, at least a bit. Maybe the tests could look at JULIA_CONDAPKG_BACKEND and modify the tests appropriately if it is Null?

Working on this now

**Comment by kescobo** [comment]
Not sure if there's a more elegant way to do this, maybe using ReTest.jl or something. But this works.

Can test locally by doing:

```
julia> using Pkg

julia> withenv("JULIA_CONDAPKG_BACKEND"=> "Null") do
           Pkg.test()
       end
```

**Comment by cjdoris** [comment]
Looks great, all your comments are fine.

Boldly merging even though there are 2 tests still running.

Thanks very much for your contribution.

**Comment by kescobo** [comment]
You're welcome! Thanks for quick review and help :-)

**Pull request merged**